common/app/routes/challenges/components/Classic-Modal.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { PropTypes } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import PureComponent from 'react-pure-render/component';
+import FontAwesome from 'react-fontawesome';
+
+export default class ClassicModal extends PureComponent {
+
+  onClickHandler() {
+    this.props.onModalClick();
+  }
```

Comment:
You don't need an instance to call a function on the props. Just pass the prop directly to the handler